uamqp/_decode.py

```diff
@@ -264,12 +266,15 @@ def decode_payload(buffer):
             except KeyError:
                 message["data"] = [value]
         elif descriptor == 118:
-            message["sequence"] = value
+            try:
+                message["sequence"].append(value)
+            except KeyError:
+                message["sequence"] = [value]
         elif descriptor == 119:
             message["value"] = value
         elif descriptor == 120:
             message["footer"] = value
-    return message
+    return Message(**message)
```

Comment:
@annatisch I'm aware of your [concern](https://github.com/Azure/azure-uamqp-python/pull/295/files#r771986720).  though for now it's private and unnecessary, I'm still in favor the idea of wrapping with a class instead of dict for the asymmetry between sending and receiving; type hint; code maintenance.

and besides the recv perf is still good with the change.
maybe if we think of improving perf in the future, we could revert the change?

Reply:
Let's keep for now - we can possible swap out the Message construct with a TypedDict for both input and output so we get the best of both.